ksql-functional-tests/src/test/resources/query-validation-tests/decimal.json

```diff
@@ -145,6 +145,183 @@
+        "CREATE STREAM TEST2 AS SELECT (a <> b) AS RESULT FROM TEST;"
+      ],
+      "inputs": [
+        {"topic": "test", "key": 0, "value": {"A": "10.01", "B":  "10.01"}},
+        {"topic": "test", "key": 0, "value": {"A": "10.01", "B":  "12.01"}},
+        {"topic": "test", "key": 0, "value": {"A": null, "B":  "10.01"}},
+        {"topic": "test", "key": 0, "value": {"A": null, "B":  null}}
+      ],
+      "outputs": [
+        {"topic": "TEST2", "key": 0, "value": {"RESULT": false}},
+        {"topic": "TEST2", "key": 0, "value": {"RESULT": true}},
+        {"topic": "TEST2", "key": 0, "value": {"RESULT": false}},
+        {"topic": "TEST2", "key": 0, "value": {"RESULT": false}}
+      ]
+    },
+    {
+      "name": "is distinct - decimal decimal",
+      "statements": [
+        "CREATE STREAM TEST (a DECIMAL(4,2), b DECIMAL(4,2)) WITH (kafka_topic='test', value_format='AVRO');",
+        "CREATE STREAM TEST2 AS SELECT (a IS DISTINCT FROM b) AS RESULT FROM TEST;"
+      ],
+      "inputs": [
+        {"topic": "test", "key": 0, "value": {"A": "10.01", "B":  "10.01"}},
+        {"topic": "test", "key": 0, "value": {"A": "10.01", "B":  "12.01"}},
+        {"topic": "test", "key": 0, "value": {"A": null, "B":  "10.01"}},
+        {"topic": "test", "key": 0, "value": {"A": null, "B":  null}}
+      ],
+      "outputs": [
+        {"topic": "TEST2", "key": 0, "value": {"RESULT": false}},
+        {"topic": "TEST2", "key": 0, "value": {"RESULT": true}},
+        {"topic": "TEST2", "key": 0, "value": {"RESULT": true}},
+        {"topic": "TEST2", "key": 0, "value": {"RESULT": false}}
+      ]
+    },
+    {
+      "name": "less than - decimal decimal",
+      "statements": [
+        "CREATE STREAM TEST (a DECIMAL(4,2), b DECIMAL(4,2)) WITH (kafka_topic='test', value_format='AVRO');",
+        "CREATE STREAM TEST2 AS SELECT (a < b) AS RESULT FROM TEST;"
+      ],
+      "inputs": [
+        {"topic": "test", "key": 0, "value": {"A": "10.01", "B":  "10.01"}},
+        {"topic": "test", "key": 0, "value": {"A": "10.01", "B":  "12.01"}},
+        {"topic": "test", "key": 0, "value": {"A": null, "B":  "10.01"}},
+        {"topic": "test", "key": 0, "value": {"A": null, "B":  null}}
+      ],
+      "outputs": [
+        {"topic": "TEST2", "key": 0, "value": {"RESULT": false}},
+        {"topic": "TEST2", "key": 0, "value": {"RESULT": true}},
+        {"topic": "TEST2", "key": 0, "value": {"RESULT": false}},
+        {"topic": "TEST2", "key": 0, "value": {"RESULT": false}}
+      ]
+    },
+    {
+      "name": "less than - decimal decimal differing scale",
+      "statements": [
+        "CREATE STREAM TEST (a DECIMAL(4,2), b DECIMAL(5,3)) WITH (kafka_topic='test', value_format='AVRO');",
+        "CREATE STREAM TEST2 AS SELECT (a < b) AS RESULT FROM TEST;"
+      ],
+      "inputs": [
+        {"topic": "test", "key": 0, "value": {"A": "10.01", "B":  "10.010"}},
```

Comment:
Out of curiosity, what would happen if these two inputs were compared with the equals operator instead? What that return true?

Reply:
Honestly, I intentionally left that vague and was going to answer that question once I have a big picture view of decimals. In the spirit of moving fast I'm going to keep it undefined (by SQL standard it looks like this might be implementation specific). It will be defined before 5.4